packages/mdc-list/foundation.js

```diff
@@ -109,29 +112,36 @@ class MDCListFoundation extends MDCFoundation {
 
   /** @param {number} index */
   setSelectedIndex(index) {
-    if (index === this.selectedIndex_) {
+    if (index < 0 && index >= this.adapter_.getListItemCount()) {
       return;
     }
 
-    const className = this.useActivatedClass_
-      ? cssClasses.LIST_ITEM_ACTIVATED_CLASS : cssClasses.LIST_ITEM_SELECTED_CLASS;
+    const prevSelectedIndex = this.selectedIndex_;
+    const hasCheckboxOrRadio = this.adapter_.hasCheckboxOrRadioAtIndex(index);
+    const ariaAttribute = hasCheckboxOrRadio ? strings.ARIA_CHECKED : strings.ARIA_SELECTED;
 
-    if (this.selectedIndex_ >= 0) {
-      this.adapter_.removeAttributeForElementIndex(this.selectedIndex_, strings.ARIA_SELECTED);
-      this.adapter_.removeClassForElementIndex(this.selectedIndex_, className);
-      this.adapter_.setAttributeForElementIndex(this.selectedIndex_, 'tabindex', -1);
+    // Update aria attributes
+    const hasCheckbox = this.adapter_.hasCheckboxAtIndex(index);
+    if (!hasCheckbox) {
+      this.adapter_.setAttributeForElementIndex(prevSelectedIndex, ariaAttribute, false);
```

Comment:
Should this be checking hasCheckboxOrRadio? Otherwise wouldn't this set the attribute even if there were no checkbox/radio in the previous item at all? (Though I'm not sure if we even expect to have lists where some items have inputs and some don't, so maybe this idea is excessive)